### PR TITLE
configurable concurrency/memory bounds

### PR DESCRIPTION
**Pull request by ransomw1c**
towards #171 , this request is about making the number of goroutines (hence the amount of memory used by the program) during upload configurable at runtime rather than statically parameterized.

### CODE REVIEW

**Comment by ransomw1c** [comment]
there'll still be some constants expressed as ratios (for example, the ratio of cafs flushes during put to the number of files uploaded at a time)

File: pkg/cafs/cafs.go [hunk 1]
```diff
@@ -154,6 +162,10 @@ func (d *defaultFs) GetAt(ctx context.Context, hash Key) (io.ReaderAt, error) {
 }
 
 func (d *defaultFs) writer(prefix string) Writer {
+	maxGoRoutines := d.concurrentFlushes
+	if maxGoRoutines < 1 {
+		maxGoRoutines = 1
```
Comment (kerneltime) [inline pkg/cafs/cafs.go]: 👍

Comment (ransomw1c) [reply pkg/cafs/cafs.go]: ^ defensive..

**Comment by kerneltime** [comment]
LGTM. A few changes

- [ ] Extend to read/download path

- [ ] Rename to concurrency/concurrency-factor? open to suggestions.

File: pkg/cafs/cafs.go (hunk 1 above)
Comment (kerneltime) [reply pkg/cafs/cafs.go]: yup

**Pull request merged**